pysat/ssnl/plot.py

```diff
@@ -38,64 +38,50 @@ def scatterplot(inst, labelx, labely, data_label, datalim, xlim=None,
     figs = []
     axs = []
 
-    # check if doing multiple data quantities
+    # Check for list-like behaviour of data_label
     if type(data_label) is str:
-        # max of one data type
+        data_label = [data_label]
+
+    # multiple data to be plotted
+    for i in np.arange(len(data_label)):
         figs.append(plt.figure())
-        ax1 = figs[0].add_subplot(211, projection='3d')
-        ax2 = figs[0].add_subplot(212)
+        ax1 = figs[i].add_subplot(211, projection='3d')
+        ax2 = figs[i].add_subplot(212)
         axs.append((ax1, ax2))
-        plt.suptitle(data_label)
+        plt.suptitle(data_label[i])
         if xlim is not None:
             ax1.set_xlim(xlim)
         ax2.set_xlim(xlim)
         if ylim is not None:
             ax1.set_ylim(ylim)
         ax2.set_ylim(ylim)
-        data_label = [data_label]
-        plt.hold(True)
-    else:
-        # multiple data to be plotted
-        for i in np.arange(len(data_label)):
-            figs.append(plt.figure())
-            ax1 = figs[i].add_subplot(211, projection='3d')
-            ax2 = figs[i].add_subplot(212)
-            axs.append((ax1, ax2))
-            # axs.append( figs[i].add_subplot(111, projection='3d') )
-            plt.suptitle(data_label[i])
-            if xlim is not None:
-                ax1.set_xlim(xlim)
-            ax2.set_xlim(xlim)
-            if ylim is not None:
-                ax1.set_ylim(ylim)
-            ax2.set_ylim(ylim)
-            plt.hold(True)
 
     # norm method so that data may be scaled to colors appropriately
     norm = colors.Normalize(vmin=datalim[0], vmax=datalim[1])
     p = [i for i in np.arange(len(figs))]
+    q = [i for i in np.arange(len(figs))]
     for i, inst in enumerate(inst):
         for j, (fig, ax) in enumerate(zip(figs, axs)):
             if len(inst.data) > 0:
```

Comment:
How about   if not inst.empty:   instead ?

Reply:
That works in the unit test.  Is this a potential solution to #130 as well?

Reply:
Maybe?

Here is the code, it uses length but on a specific variable.

@property
    def empty(self):
        """Boolean flag reflecting lack of data.
        True if there is no Instrument data."""

        if self.pandas_format:
            return self.data.empty
        else:
            if 'time' in self.data.indexes:
                return len(self.data.indexes['time']) == 0
            else:
                return True

Reply:
Looks like a straightforward replace.  May need to update the pandas_format=False case to look for
```
len(ds.variables)==0
```
which seems to be what xarray prefers.  This should be more generalized, ie, `time` does not have to exist.

Reply:
created a new branch to see if this works.

Reply:
The short answer is no.  :(
The data is not loaded to self until the end of the function, so the property cannot be used in Line 940 as the property does not yet exist.  But I think I found another way.

Reply:
Time does have to exist, it is part of pysat's requirements on data. 

Replacing:
if len(inst.data) > 0:
with:
if not inst.empty:

doesn't work?

Reply:
The `len(ds.variables) == 0` broke a bunch of other things.  I've left it as is.  

A simple replace doesn't work because the data object is a pure pandas or xarray object, not a pysat instrument, hence empty is not defined for xarray.  Check out Pull #199 for the fix.

Reply:
It is supposed to be a pysat Instrument. I do see that part is missing from the docstring.